src/Component/Tests/Symfony/EventListener/ReadListenerTest.php

```diff
@@ -0,0 +1,167 @@
+use Sylius\Resource\Context\Initiator\RequestContextInitiatorInterface;
+use Sylius\Resource\Metadata\Create;
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\Operation\HttpOperationInitiatorInterface;
+use Sylius\Resource\State\ProviderInterface;
+use Sylius\Resource\Symfony\EventListener\ReadListener;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class ReadListenerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private HttpOperationInitiatorInterface|ObjectProphecy $operationInitiator;
+
+    private RequestContextInitiatorInterface|ObjectProphecy $contextInitiator;
+
+    private ProviderInterface|ObjectProphecy $provider;
+
+    private ReadListener $readListener;
+
+    protected function setUp(): void
+    {
+        $this->operationInitiator = $this->prophesize(HttpOperationInitiatorInterface::class);
+        $this->contextInitiator = $this->prophesize(RequestContextInitiatorInterface::class);
+        $this->provider = $this->prophesize(ProviderInterface::class);
+
+        $this->readListener = new ReadListener(
+            $this->operationInitiator->reveal(),
+            $this->contextInitiator->reveal(),
+            $this->provider->reveal(),
+        );
+    }
+
+    /** @test */
+    public function it_retrieves_data_and_store_them_to_request(): void
+    {
+        $event = $this->prophesize(RequestEvent::class);
+        $request = $this->prophesize(Request::class);
+        $attributes = $this->prophesize(ParameterBag::class);
+        $operation = $this->prophesize(HttpOperation::class);
+
+        $event->getRequest()->willReturn($request);
+
+        $context = new Context();
+
+        $this->contextInitiator->initializeContext($request)->willReturn($context);
+
+        $this->operationInitiator->initializeOperation($request)->willReturn($operation);
+
+        $request->attributes = $attributes;
+
+        $this->provider->provide($operation, Argument::type(Context::class))->willReturn(['foo' => 'fighters'])->shouldBeCalled();
+
+        $attributes->set('data', ['foo' => 'fighters'])->shouldBeCalled();
+
+        $this->readListener->onKernelRequest($event->reveal());
+    }
+
+    /** @test */
+    public function it_does_nothing_when_operation_is_a_create_operation(): void
+    {
+        $event = $this->prophesize(RequestEvent::class);
+        $request = $this->prophesize(Request::class);
+        $attributes = $this->prophesize(ParameterBag::class);
+
+        $event->getRequest()->willReturn($request);
+
+        $operation = new Create();
+
+        $context = new Context();
+
+        $this->contextInitiator->initializeContext($request)->willReturn($context);
+
+        $this->operationInitiator->initializeOperation($request)->willReturn($operation);
+
+        $request->attributes = $attributes;
```

Comment:
Is it possible or makes sense to prophesize attributes directly to `$request->attributes`?

Reply:
nope, we cannot do that.

Reply:
We also did like that on the old spec file.